### PR TITLE
 [SYCL] Update tests with -fno-sycl-id-queries-fit-in-int option

### PR DESCRIPTION
**Pull request by smanna12**
The changes are due to the PR: intel/llvm#3427

Signed-off-by: Soumi Manna soumi.manna@intel.com

### CODE REVIEW

**Comment by smanna12** [comment]
The failures seem unrelated to my change.

File: SYCL/SubGroup/sub_group_as.cpp
Comment (AlexeySachkov) [inline SYCL/SubGroup/sub_group_as.cpp:1]: I wonder, why those changes are needed at all?

Comment (smanna12) [reply SYCL/SubGroup/sub_group_as.cpp:1]: Thanks @AlexeySachkov for reviewing the patch. 

https://github.com/intel/llvm/pull/3427: Enables the option -fsycl-id-queries-fit-in-int to true by default.  sub_group_as.cpp passes when the option is false on default mode but fails with default ON. So I added -fno-sycl-id-queries-fit-in-int to disable the option and match with the old behavior. 

 sycl-id-queries-fit-in-int  makes a big impact on performance for DPC++ in many cases. The test failures currently block https://github.com/intel/llvm/pull/3427. Is there any workaround i can use here? Thanks.

Comment (vladimirlaz) [reply SYCL/SubGroup/sub_group_as.cpp:1]: @smanna12, please create issue for the regression. It looks like the test start to fail after changing default behavior.
In short: load/store functions for local pointer stop work as expected.

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Comment by smanna12** [comment]
>>@smanna12, please create issue for the regression. It looks like the test start to fail after changing default behavior.
In short: load/store functions for local pointer stop work as expected.

@vladimirlaz, I have created https://github.com/intel/llvm/issues/3495 for the regression and assigned to you. Thanks.

**Review by vladimirlaz — APPROVED** [review]

File: SYCL/SubGroup/sub_group_as.cpp
Comment (smanna12) [reply SYCL/SubGroup/sub_group_as.cpp:1]: @AlexeySachkov, could you please review this?

**Comment by smanna12** [comment]
> > > @smanna12, please create issue for the regression. It looks like the test start to fail after changing default behavior.
> > > In short: load/store functions for local pointer stop work as expected.
> 
> @vladimirlaz, I have created [intel/llvm#3495](https://github.com/intel/llvm/issues/3495) for the regression and assigned to you. Thanks.

@@vladimirlaz, Any update on this? Thanks. Blocks PR: intel/llvm#3427

**Review by AlexeySachkov — APPROVED** [review]
I'm okay to submit this patch since we have an issue filed to fix the regression

**Comment by smanna12** [comment]
Thanks @AlexeySachkov for the review. @vladimirlaz, could you please commit it?. Thanks.

File: SYCL/SubGroup/sub_group_as.cpp
Comment (vladimirlaz) [reply SYCL/SubGroup/sub_group_as.cpp:1]: It looks like root cause for failing test is the same with post-commit failures after merging intel/llvm#3427

Comment (smanna12) [reply SYCL/SubGroup/sub_group_as.cpp:1]: > It looks like root cause for failing test is the same with post-commit failures after merging [intel/llvm#3427](https://github.com/intel/llvm/pull/3427)

@vladimirlaz, Does it mean that the test needs to be updated? Not sure i understand your comments.

Comment (smanna12) [reply SYCL/SubGroup/sub_group_as.cpp:1]: @vladimirlaz, PR: intel/llvm#3427: which has not been merged yet. I am waiting for this PR to be merged first before merging PR: intel/llvm#3427

**Pull request merged**
